src/utils/instances.js

```diff
@@ -66,7 +86,7 @@ export class NodeInstance {
   }
 
   mapCalculate (calculate) {
-    return async component => {
+    return async () => {
```

Comment:
This is pretty much what this whole pull request is for. `calculate` now works independently of the vue component with the vue component wrapping the instance and not the other way around. All the necessary data for computation is kept inside the instance.